libs/rollup.config.js

```diff
@@ -23,43 +34,131 @@ const commonConfig = {
     resolve({ extensions, preferBuiltins: true }),
     typescript()
   ],
-  external: [
-    ...Object.keys(pkg.dependencies),
-    ...Object.keys(pkg.devDependencies),
-    'ethereumjs-util',
-    'ethereumjs-wallet',
-    'ethers/lib/utils',
-    'ethers/lib/index',
-    'hashids/cjs',
-  ]
+  external
+}
+
+// For the browser bundle, these need to be internal because they either:
+// * contain deps that need to be polyfilled via `nodePolyfills`
+// * are ignored via `ignore`
+const internal = [
+  'eth-sig-util',
+  'ethereumjs-tx',
+  'ethereumjs-util',
+  'ethereumjs-wallet',
+  'graceful-fs',
+  'node-localstorage',
+  'web3'
+]
+
+const browserConfig = {
+  plugins: [
+    ignore(['web3', 'graceful-fs', 'node-localstorage']),
+    resolve({ extensions, preferBuiltins: false }),
+    commonjs({
+      extensions,
+      transformMixedEsModules: true,
+      dynamicRequireTargets: [
+        'data-contracts/ABIs/*.json',
+        'eth-contracts/ABIs/*.json'
+      ]
+    }),
+    alias({
+      entries: [{ find: 'stream', replacement: 'stream-browserify' }]
+    }),
+    nodePolyfills(),
```

Comment:
so this is helps for the 3rd parties that assume node env?

Reply:
Yeah for example webpack 5, which is used by newest CRA, doesn't polyfill node. But I didn't want people to have to use craco or eject just to use our sdk. It also presents a problem because things can only be polyfilled in internal (bundled) deps, so bundle size quickly spirals out of control if we include libs in this entrypoint

Reply:
cool lots to think about here, great stuff